SU-SWS/stanford_ssp
# SITES-616: Add workgroup API role mapping option

# READY FOR REVIEW

# Summary
- Adds workgroup API role evaluation
- Adds ability to toggle between workgroup formats ":" vs "_"

# Needed By (Date)
- End of sprint?

# Urgency
- 6 out of 10?

# Steps to Test

1. On your local, check out the latest of the socorro branch for acsf_cardinald7 and this branch
2. Ensure you have a jsv or other jumpstart site with the `acsf` itasks build so that stanford_ssp module is enabled
3. Log in as an administrator and navigate to /admin/config/stanford/stanford_ssp/role-mappings
4. Add an administrator workgroup mapping for itservices:webservices
5. Select workgroup api radio as the validation source
6. Enter the server path to the cert `buws-cardinald7-test.cert`
7. Enter the server path to the key `wkgrp-test.key`
8. Manually set the workgroup api to the UAT server via drush `drush vset stanford_ssp_workgroup_api_endpoint https://workgroupsvc-uat.stanford.edu/v1/workgroups/`
9. Remove all of your users roles through the UI
10. Log out of your user
11. Log back in via `sso/login`
12. Validate you have new roles

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
